### ISSUE TITLE
Avoid matching `\iff` in syntax highlighting comment matching

### ISSUE THREAD

**Issue by user202729** [lervag/vimtex#2748]
Basically currently if there's some `\iffalse ... \fi` it will highlight the block as a comment, but `\iffalse $ a \iff b $ \fi more content` will also highlight `more content` as comment because it thought `\iff` needs to be matched.

Remark: generally TeX is not deterministically parsable anyway, but `\iff` is a very common case. Looking around, some less common ones are `\ifcurrfiledir` etc. (currfile package) or `ifnot@nil` (macros2e) .

**Comment by lervag** [comment]
Agh, yes, TeX is hard to parse! But in this case, I believe it should not be too hard - we just need to recognize that `\iff` does not start a new `\if ... \fi` group.

**Comment by lervag** [comment]
I believe this is fixed now.
